src/taoensso/sente.cljc

```diff
@@ -1043,7 +1048,9 @@
                                  (or
                                    (when (handshake? clj)
                                      (receive-handshake! :ws chsk clj)
-                                     (reset! retry-count_ 0))
+                                     (reset! retry-count_ 0)
+                                     (swap-chsk-state! chsk
```

Comment:
Please note that `swap-chsk-state!` can return falsey, so this form needs to be followed up with a `true` to prevent possible interference with the enclosing `or`